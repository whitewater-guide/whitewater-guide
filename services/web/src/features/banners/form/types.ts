import {
  BannerFormInput,
  Group,
  Region,
  UploadLink,
} from '@whitewater-guide/commons';
import { RouteComponentProps } from 'react-router';
import { InjectedFormProps } from 'redux-form';

export type BannerFormProps = InjectedFormProps<BannerFormInput> &
  RouteComponentProps<any> & {
    groups: Group[];
    regions: Region[];
    uploadLink: UploadLink;
  };
