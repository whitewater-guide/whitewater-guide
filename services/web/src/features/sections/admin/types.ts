export interface RouterParams {
  sectionId: string;
}
