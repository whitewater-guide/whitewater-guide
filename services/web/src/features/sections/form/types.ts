import {
  Region,
  River,
  SectionFormInput,
  Tag,
} from '@whitewater-guide/commons';
import { MdEditorValue } from '@whitewater-guide/md-editor';
import { RouteComponentProps } from 'react-router';
import { InjectedFormProps } from 'redux-form';

export type SectionFormData = SectionFormInput<MdEditorValue>;

export interface SectionFormProps
  extends InjectedFormProps<SectionFormData>,
    RouteComponentProps<any> {
  region: Region;
  river: River;
  tags: Tag[];
}
