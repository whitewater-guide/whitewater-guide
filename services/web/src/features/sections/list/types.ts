import { WithSectionsList } from '@whitewater-guide/clients';
import { RouteComponentProps } from 'react-router-dom';
import { WithDeleteMutation } from '../../../apollo';

export type SectionsListProps = WithSectionsList &
  WithDeleteMutation<'removeSection'> &
  RouteComponentProps<{ regionId: string }>;
