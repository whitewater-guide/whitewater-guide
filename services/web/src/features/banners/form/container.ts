import { withFeatureIds } from '@whitewater-guide/clients';
import {
  BannerFormStruct,
  BannerInput,
  BannerKind,
  BannerPlacement,
} from '@whitewater-guide/commons';
import { compose, mapProps } from 'recompose';
import { formContainer } from '../../../components/forms';
import BANNER_FORM_QUERY from './bannerForm.query';
import deserializeBannerForm from './deserializeBannerForm';
import serializeBannerForm from './serializeBannerForm';
import { BannerFormProps } from './types';
import UPSERT_BANNER from './upsertBanner.mutation';

const NEW_BANNER: BannerInput = {
  id: null,
  slug: '',
  name: '',
  priority: 0,
  enabled: true,
  placement: BannerPlacement.MOBILE_REGION_DESCRIPTION,
  source: {
    kind: BannerKind.Image,
    ratio: null,
    src: null,
  },
  link: null,
  extras: null,
  regions: [],
  groups: [],
};

const bannerForm = formContainer({
  formName: 'banner',
  propName: 'banner',
  defaultValue: NEW_BANNER,
  query: BANNER_FORM_QUERY,
  mutation: UPSERT_BANNER,
  serializeForm: serializeBannerForm,
  deserializeForm: deserializeBannerForm,
  validationSchema: BannerFormStruct,
});

export default compose<BannerFormProps, {}>(
  withFeatureIds('banner'),
  bannerForm,
  mapProps(({ data, ...props }: any) => ({
    regions: data.regions.nodes,
    groups: data.groups.nodes,
    uploadLink: data.uploadLink,
    ...props,
  })),
);
