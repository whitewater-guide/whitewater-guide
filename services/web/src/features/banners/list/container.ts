import { queryResultToList, WithList } from '@whitewater-guide/clients';
import { Banner, Connection } from '@whitewater-guide/commons';
import { graphql } from 'react-apollo';
import { compose } from 'recompose';
import { withDeleteMutation } from '../../../apollo';
import { withLoading } from '../../../components';
import LIST_BANNERS from './listBanners.query';
import REMOVE_BANNER from './removeBanner.mutation';
import { BannerListProps } from './types';

interface Result {
  banners: Connection<Banner>;
}

export interface WithBannersList {
  banners: WithList<Banner>;
}

const withBannersList = graphql<any, Result, any, WithBannersList>(
  LIST_BANNERS,
  {
    alias: 'withBannersList',
    options: () => ({
      fetchPolicy: 'cache-and-network',
    }),
    props: (props) => queryResultToList(props, 'banners'),
  },
);

export default compose<BannerListProps, {}>(
  withBannersList,
  withDeleteMutation({
    mutation: REMOVE_BANNER,
    propName: 'removeBanner',
    refetchQueries: ['listBanners'],
  }),
  withLoading<WithBannersList>(({ banners }) => banners.loading),
);
