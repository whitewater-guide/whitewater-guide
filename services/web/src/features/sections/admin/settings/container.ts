import { withFeatureIds } from '@whitewater-guide/clients';
import {
  Section,
  SectionAdminSettings,
  SectionAdminSettingsStruct,
} from '@whitewater-guide/commons';
import { withRouter } from 'react-router';
import { compose } from 'recompose';
import { InjectedFormProps } from 'redux-form';
import { formContainer, serializeForm } from '../../../../components/forms';
import ADMINISTRATE_SECTION_MUTATION from './administrateSection.mutation';
import SECTION_ADMIN_SETTINGS_QUERY from './sectionAdmin.query';

const sectionAdminForm = formContainer({
  formName: 'sectionAdminSettings',
  propName: 'settings',
  defaultValue: {},
  query: SECTION_ADMIN_SETTINGS_QUERY,
  mutation: ADMINISTRATE_SECTION_MUTATION,
  serializeForm: serializeForm(),
  deserializeForm: ({ demo }: Section) => ({ demo }),
  validationSchema: SectionAdminSettingsStruct,
  extraVariables: ({ sectionId }) => ({ sectionId }),
  backPath: null,
});

export default compose<InjectedFormProps<SectionAdminSettings>, {}>(
  withRouter,
  withFeatureIds('section'),
  sectionAdminForm,
);
