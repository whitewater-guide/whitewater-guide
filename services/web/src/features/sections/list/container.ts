import { WithSectionsList } from '@whitewater-guide/clients';
import { withRouter } from 'react-router-dom';
import { compose } from 'recompose';
import { withDeleteMutation } from '../../../apollo';
import REMOVE_SECTION from './removeSection.mutation';
import { SectionsListProps } from './types';

export default compose<SectionsListProps, WithSectionsList>(
  withRouter,
  // withSectionsList(),
  withDeleteMutation({
    mutation: REMOVE_SECTION,
    propName: 'removeSection',
    refetchQueries: ['listSections'],
  }),
);
