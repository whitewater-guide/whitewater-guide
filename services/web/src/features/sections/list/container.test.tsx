import { flushPromises } from '@whitewater-guide/clients';
import casual from 'casual';
import { ReactWrapper } from 'enzyme';
import React from 'react';
import { Loading } from '../../../components';
import { FormReceiver, mountForm } from '../../../test';
import container from './container';
import { SectionsListProps } from './types';

let wrapped: ReactWrapper;
let receiver: ReactWrapper<SectionsListProps>;

beforeEach(async () => {
  casual.seed(1);
});

afterEach(() => {
  wrapped.unmount();
});

const removeSection = jest.fn(() => 'deleted');

const mount = (regionId?: string) => {
  wrapped = mountForm({
    form: container,
    props: { regionId },
    mockApollo: true,
    mutations: { removeSection },
  });
};

it('should have region id', async () => {
  mount('lol');
  await flushPromises();
  wrapped.update();
  receiver = wrapped.find(FormReceiver).first() as any;
  expect(receiver.prop('regionId')).toBe('lol');
});

it('should have loading state', async () => {
  mount();
  expect(wrapped.containsMatchingElement(<Loading />)).toBe(true);
  await flushPromises();
  wrapped.update();
  expect(wrapped.containsMatchingElement(<Loading />)).toBe(false);
});

it('should delete section', async () => {
  mount();
  await flushPromises();
  wrapped.update();
  receiver = wrapped.find(FormReceiver).first() as any;
  receiver.prop('removeSection')('foo');
  await flushPromises();
  expect(removeSection).toBeCalled();
});

it('should match snapshot', async () => {
  mount();
  await flushPromises();
  wrapped.update();
  receiver = wrapped.find(FormReceiver).first() as any;
  expect(receiver.prop('sections')).toMatchSnapshot();
});
