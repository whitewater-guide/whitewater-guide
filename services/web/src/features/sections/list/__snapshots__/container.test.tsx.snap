// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
Object {
  "count": -603,
  "error": undefined,
  "fetchMore": [Function],
  "loading": false,
  "nodes": Array [
    Object {
      "__typename": "Section",
      "altNames": Array [
        "Omnis et",
        "Delectus et ipsam",
      ],
      "difficulty": 0.1300285691395402,
      "difficultyXtra": "Voluptatem dicta cum",
      "distance": 0.9033701713196933,
      "drop": 0.28777534305118024,
      "duration": -761,
      "flows": Object {
        "__typename": "GaugeBinding",
        "approximate": false,
        "impossible": 0.7173916019964963,
        "maximum": 0.28976000170223415,
        "minimum": 0.8006326730828732,
        "optimum": 0.23297427338548005,
        Symbol(id): "$Section:7fbe024f-3316-4265-a6e8-c65a837e308a:Qui itaque.flows",
      },
      "flowsText": "Iure molestiae eos",
      "gauge": Object {
        "__typename": "Gauge",
        "flowUnit": "Quod non",
        "id": "df1759f6-73ce-4c94-8c92-4980eb57ec97",
        "language": "Enim non",
        "lastFlow": 0.9695957456715405,
        "lastLevel": 0.5700666578486562,
        "lastTimestamp": 2017-01-31T21:00:00.000Z,
        "levelUnit": "Quia rem",
        "location": Object {
          "__typename": "Point",
          "coordinates": Array [
            0.09155644034035504,
            0.5857592679094523,
          ],
          "id": "2b713c8c-2e0b-4812-8601-1831d4838fd4",
          "kind": "Ut minima",
          Symbol(id): "Point:2b713c8c-2e0b-4812-8601-1831d4838fd4",
        },
        "name": "Est omnis asperiores",
        Symbol(id): "Gauge:df1759f6-73ce-4c94-8c92-4980eb57ec97:Enim non",
      },
      "id": "7fbe024f-3316-4265-a6e8-c65a837e308a",
      "language": "Qui itaque",
      "levels": Object {
        "__typename": "GaugeBinding",
        "approximate": false,
        "impossible": 0.018647290067747235,
        "maximum": 0.5610302197746933,
        "minimum": 0.41792653826996684,
        "optimum": 0.36784324049949646,
        Symbol(id): "$Section:7fbe024f-3316-4265-a6e8-c65a837e308a:Qui itaque.levels",
      },
      "name": "Esse blanditiis natus",
      "putIn": Object {
        "__typename": "Point",
        "coordinates": Array [
          0.792403579223901,
          0.9445947585627437,
        ],
        "id": "2e447909-892f-494b-8186-eb7680c80a28",
        "kind": "Qui aut",
        Symbol(id): "Point:2e447909-892f-494b-8186-eb7680c80a28",
      },
      "rating": 0.9168613343499601,
      "river": Object {
        "__typename": "River",
        "altNames": Array [
          "Veniam ducimus",
          "Modi odit architecto",
        ],
        "id": "0b8eded1-c042-40ea-896a-e683b452acd6",
        "language": "Aperiam placeat",
        "name": "Maiores distinctio placeat",
        Symbol(id): "River:0b8eded1-c042-40ea-896a-e683b452acd6:Aperiam placeat",
      },
      "season": "Totam et",
      "seasonNumeric": Array [
        -808,
        -413,
      ],
      "shape": Array [
        Array [
          0.49810906942002475,
          0.6216957198921591,
        ],
        Array [
          0.07379201101139188,
          0.11474596988409758,
        ],
      ],
      "takeOut": Object {
        "__typename": "Point",
        "coordinates": Array [
          0.752788350218907,
          0.428091190289706,
        ],
        "id": "f8272ec6-782a-4e55-acab-3eb9ac55c45e",
        "kind": "Sapiente perferendis nam",
        Symbol(id): "Point:f8272ec6-782a-4e55-acab-3eb9ac55c45e",
      },
      "updatedAt": 2017-01-31T21:00:00.000Z,
      Symbol(id): "Section:7fbe024f-3316-4265-a6e8-c65a837e308a:Qui itaque",
    },
    Object {
      "__typename": "Section",
      "altNames": Array [
        "Quis ut ab",
        "Quod labore ullam",
      ],
      "difficulty": 0.9310129284858704,
      "difficultyXtra": "In cupiditate quo",
      "distance": 0.8047545654699206,
      "drop": 0.14326300681568682,
      "duration": -613,
      "flows": Object {
        "__typename": "GaugeBinding",
        "approximate": true,
        "impossible": 0.43024727678857744,
        "maximum": 0.6219193686265498,
        "minimum": 0.13019471801817417,
        "optimum": 0.5614589587785304,
        Symbol(id): "$Section:db686250-31e0-4e1c-89b0-9a0214f59431:Adipisci itaque.flows",
      },
      "flowsText": "Quam cum nobis",
      "gauge": Object {
        "__typename": "Gauge",
        "flowUnit": "Sint in",
        "id": "c0ef7685-d942-47f9-90f9-263cd53ed982",
        "language": "Repellendus officia",
        "lastFlow": 0.24021763261407614,
        "lastLevel": 0.3821027507074177,
        "lastTimestamp": 2017-01-31T21:00:00.000Z,
        "levelUnit": "Ab cum",
        "location": Object {
          "__typename": "Point",
          "coordinates": Array [
            0.45087393489666283,
            0.35981619404628873,
          ],
          "id": "1dfc64f1-82d3-496a-94b9-ac1d0ccbed05",
          "kind": "Libero totam",
          Symbol(id): "Point:1dfc64f1-82d3-496a-94b9-ac1d0ccbed05",
        },
        "name": "Ab illo sit",
        Symbol(id): "Gauge:c0ef7685-d942-47f9-90f9-263cd53ed982:Repellendus officia",
      },
      "id": "db686250-31e0-4e1c-89b0-9a0214f59431",
      "language": "Adipisci itaque",
      "levels": Object {
        "__typename": "GaugeBinding",
        "approximate": true,
        "impossible": 0.531668106559664,
        "maximum": 0.02316873031668365,
        "minimum": 0.41081135300919414,
        "optimum": 0.40147958556190133,
        Symbol(id): "$Section:db686250-31e0-4e1c-89b0-9a0214f59431:Adipisci itaque.levels",
      },
      "name": "Sequi quod",
      "putIn": Object {
        "__typename": "Point",
        "coordinates": Array [
          0.4859906700439751,
          0.16695159254595637,
        ],
        "id": "8451bb8c-ae2e-4a06-8b9c-e6fe83e0ae64",
        "kind": "Accusantium unde qui",
        Symbol(id): "Point:8451bb8c-ae2e-4a06-8b9c-e6fe83e0ae64",
      },
      "rating": 0.9248079678509384,
      "river": Object {
        "__typename": "River",
        "altNames": Array [
          "Et architecto",
          "Aspernatur facere",
        ],
        "id": "639fde2c-1a10-47d9-b975-9f1926883bea",
        "language": "Ut dolores",
        "name": "Voluptas quis",
        Symbol(id): "River:639fde2c-1a10-47d9-b975-9f1926883bea:Ut dolores",
      },
      "season": "Quasi ex",
      "seasonNumeric": Array [
        -479,
        -809,
      ],
      "shape": Array [
        Array [
          0.5820141793228686,
          0.8962802486494184,
        ],
        Array [
          0.8788319895975292,
          0.24148913635872304,
        ],
      ],
      "takeOut": Object {
        "__typename": "Point",
        "coordinates": Array [
          0.5343773367349058,
          0.3447366547770798,
        ],
        "id": "fce569f0-3521-42b8-807f-ed50721582ec",
        "kind": "Itaque ex",
        Symbol(id): "Point:fce569f0-3521-42b8-807f-ed50721582ec",
      },
      "updatedAt": 2017-01-31T21:00:00.000Z,
      Symbol(id): "Section:db686250-31e0-4e1c-89b0-9a0214f59431:Adipisci itaque",
    },
  ],
  "refetch": [Function],
}
`;
